gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -309,9 +322,22 @@ public void start() {
 
     // Start metric reporting
     if (this.containerMetrics.isPresent()) {
-      this.containerMetrics.get()
-          .startMetricReportingWithFileSuffix(ConfigUtils.configToState(this.clusterConfig),
-              this.taskRunnerId);
+      try {
+        this.containerMetrics.get()
+            .startMetricReportingWithFileSuffix(ConfigUtils.configToState(this.clusterConfig), this.taskRunnerId);
+      } catch (MetricReporterException e) {
+        logger.error("Failed to start {} metric reporter", e.getType().name(), e);
+        if (this.isMetricReportingFailureFatal) {
+          logger.error("Failing TaskRunner since metric reporting failure is FATAL.");
+          Throwables.propagate(e);
```

Comment:
Shall we avoid logging + throwing exception if failure is fatal? This might make the stack trace hard to read.

Reply:
Addressed.